src/main/java/ca/wescook/nutrition/data/NutrientManager.java

```diff
@@ -38,12 +40,24 @@ public void set(Nutrient nutrient, Float value) {
         nutrition.put(nutrient, value);
     }
 
-    public void vampireFloor() {
+    /**
+     *
+     * @return true if any nutrition change
+     */
+    public boolean setVampireFoodAmount(ExtendedPlayer extendedPlayer) {
+        float maxNutrition = Config.vampireMinNutrition
+            + (Config.vampireMaxNutrition - Config.vampireMinNutrition) * extendedPlayer.getVampireLevel() / 10;
+        float nutritionValue = ((float) extendedPlayer.getBloodPower() / extendedPlayer.getMaxBloodPower())
+            * maxNutrition;
+
+        boolean change = false;
         for (Nutrient nutrient : NutrientList.get()) {
-            if (nutrition.get(nutrient) < 50) {
-                nutrition.put(nutrient, 50f);
+            if (nutrition.get(nutrient) != nutritionValue) {
```

Comment:
As you are playing with floats calculations in both ways, I highly recommend you to round it up while comparing to avoid issues:

_(also require to import Math)_
```suggestion
            if (Math.abs(nutrition.get(nutrient) - nutritionValue) > 0.01f) {
```